from math import sin, cos, pi
import rclpy
from rclpy.node import Node
from rclpy.qos import QoSProfile
from geometry_msgs.msg import Quaternion
from sensor_msgs.msg import JointState
from trajectory_msgs.msg import JointTrajectory, JointTrajectoryPoint
from builtin_interfaces.msg import Duration
from tf2_ros import TransformBroadcaster, TransformStamped
import numpy as np
import time
from ament_index_python.packages import get_package_share_directory
import os, sys
helper_path = os.path.join(
            get_package_share_directory('prairie_control'),
            "helpers")

sys.path.append(helper_path)
import helpers

print(helper_path, dir(helpers))
#JOINT_LIST_COMPLETE, _, _ = helpers.makeJointList()
JOINT_LIST_COMPLETE = ["l_hip_pitch", "l_hip_roll", "l_hip_yaw", "l_knee", "l_foot_pitch", "l_foot_roll",
                       "r_hip_pitch", "r_hip_roll", "r_hip_yaw", "r_knee", "r_foot_pitch", "r_foot_roll"]

class default_state(Node):

    def __init__(self):
        rclpy.init()
        super().__init__('state_publisher')

        qos_profile = QoSProfile(depth=10)
        self.joint_pub = self.create_publisher(JointState, 'joint_states', qos_profile)
        self.broadcaster = TransformBroadcaster(self, qos=qos_profile)
        self.nodeName = self.get_name()
        self.get_logger().info("{0} started".format(self.nodeName))

        loop_rate = self.create_rate(30)
        joint_state = JointState()
        joint_traj = JointTrajectory()
        jtp = JointTrajectoryPoint()
        duration = Duration()
        try:
            while rclpy.ok():
                rclpy.spin_once(self)

                # update joint_state
                now = self.get_clock().now()
                joint_state.header.stamp = now.to_msg()
                joint_state.name = JOINT_LIST_COMPLETE
                joint_state.position = [0. ] * len(JOINT_LIST_COMPLETE)
                self.joint_pub.publish(joint_state)

                loop_rate.sleep()

        except KeyboardInterrupt:
            pass

def main():
    node = default_state()

if __name__ == '__main__':
    main()